src/index.ts

```diff
@@ -49,10 +49,9 @@ export const recoilPersist = (
       if (
         newValue !== null &&
         newValue !== undefined &&
-        newValue instanceof DefaultValue &&
-        state.hasOwnProperty(node.key)
+        newValue instanceof DefaultValue
       ) {
-        delete state[node.key]
+        if (state.hasOwnProperty(node.key)) delete state[node.key]
```

Comment:
Looks like these two construction are the same:

```
if (a && b) {
}
```

and 

```
if (a) {
   if (b) {
   }
}
```

Reply:
They are not the same. If the state does not contain the node.key it will still persist the DefaultValue object because then it will fall in the else branch.